### ISSUE TITLE
Presenter display layout hosed on Firefox 9 and 10

### ISSUE THREAD

**Issue by durin42** [schacon/showoff#167]
I did some cursory investigation, but I'm honestly not sure what's up with the layout. It appears to be some disagreement between Webkit and Gecko on how to render the slide div. Seeing this with ffox on OS X, but I'd imagine it happens on other platforms as well.

**Comment by durin42** [comment]
I'm not seeing this as entirely fixed - the div containing the presentation is still 1024x768 despite being visually scaled down by 50%. The net result is that the speaker notes are pushed off the bottom of my screen on my MacBook Air.

**Comment by goncalossilva** [comment]
This is not a fix for the problem, but for the compatibility issues around FF. It's now using the same hack as in Chrome.

The presenter view definitely needs some love.

**Comment by durin42** [comment]
Should I open a new issue reflecting the current state of the presenter view in ff?

**Comment by goncalossilva** [comment]
Does the same issue (notes being pushed off) happen on Chrome/Safari? If yes, please open a new issue. If not, please reopen this one :-)

**Comment by durin42** [comment]
It does not occur in Chrome or Safari. I don't have permission to reopen issues, so I can't do so.

**Comment by goncalossilva** [comment]
Thanks for digging this for me!

**Comment by bknowles** [comment]
I also noticed problems in presenter mode on Firefox.  Is there anything I can do to help debug this issue?

**Comment by goncalossilva** [comment]
The same problems @durin42 mentioned?

**Comment by bknowles** [comment]
I believe so.  Do you want to see some screencaps?

**Comment by goncalossilva** [comment]
That would be helpful. That and some CSS fu :p

**Comment by bknowles** [comment]
Dunno that I can help with the CSS fu, but I can give you some screencaps.  These come from doing a showoff presentation of my tutorial at https://github.com/bknowles/chef-fundamentals/tree/flossuk-EDI-2012 which I am preparing to present on the 20th of March in Edinburgh.  It shows up okay in Chrome, so while I have other problems with showoff, this particular one isn't a show stopper for me right now.

Anyway, let me now go try to figure out how to upload the screencaps.

**Comment by bknowles** [comment]
Screencaps uploaded to https://github.com/bknowles/showoff/tree/issue-167/screencaps

**Comment by durin42** [comment]
Weird, that's not quite what I'm seeing with master. I'll try to post screenshots later today.

**Comment by goncalossilva** [comment]
@bknowles are you sure you're testing off master? I committed a work-around to prevent some of your issues on FF: https://github.com/schacon/showoff/commit/a92d2e4e6d1fe6fbceb6bdb1ab8cc4b67577d388

**Comment by bknowles** [comment]
No, I was testing off the released gem for showoff-0.7.0.

If you can give me some pointers on how to install the current "master" version with all associated dependencies, I'll be glad to re-do my tests and see what comes up.

**Comment by goncalossilva** [comment]
Sure!

Edit your Gemfile and replace your showoff line with this one: `gem "showoff", :git => "git://github.com/schacon/showoff.git"`

Run `bundle install` and `bundle exec showoff serve` ;)

**Comment by bknowles** [comment]
So, this is the relevant current contents of /usr/bin/showoff:

gem 'showoff', version
load Gem.bin_path('showoff', 'showoff', version)

Can I just replace these two lines with your one line?  I'll try that, but I wanted to ask to be sure.

**Comment by goncalossilva** [comment]
No, no! The file you need to edit is the Gemfile, located in the root of your presentation folder.

**Comment by bknowles** [comment]
The only Gemfile that I have anywhere remotely close to my presentation folder is in the showoff subdirectory that was installed when I forked the showoff repository, so that I could upload my screencaps.  I made sure to keep that subdirectory close to the directories where I was doing the presentations, so that I could try to keep all that related work together in my directory structure.

I've been running "showoff serve" manually from the command line, when I was in the appropriate /slides subdirectory of the presentation I was working on -- either from the subdirectory corresponding to https://github.com/bknowles/zero-to-cloud/tree/master/slides or from the subdirectory corresponding to https://github.com/bknowles/chef-fundamentals/tree/master/slides

**Comment by goncalossilva** [comment]
Ok, one of your presentations has a Gemfile: https://github.com/bknowles/chef-fundamentals/blob/master/Gemfile

Just change the line I wrote before, run `bundle install`, and use `bundle exec showoff serve`.

**Comment by bknowles** [comment]
Aw, crap.  I completely missed that Gemfile.  And I hadn't copied that over to my other project.  Fixing that now.

Let me try the change and see if this fixes stuff, at least with regards to how I presume things should look when using the official version of showoff, instead of the fork that jtimberman created.  ;-(

**Comment by bknowles** [comment]
Okay, well this looks different with Firefox.  New screencaps to be uploaded shortly.

**Comment by bknowles** [comment]
Okay, it is still borked, but now it's borked in a somewhat different way.  See https://github.com/bknowles/showoff/tree/issue-167/screencaps

**Comment by durin42** [comment]
This is exactly the behavior I'm seeing. I fiddled with firebug a bit, and the div containing the slides (I think it's class="content"?) had a height of 768px despite the transform being applied.

**Comment by goncalossilva** [comment]
Yeah, it's pretty annoying. FF doesn't implement "zoom:" ATM. I'll try to think of a better work-around.
